projects/chaos/inc/power_path.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+// Chaos Power Path monitoring module.
+// Requires gpio, gpio interrupts, adcs, soft timers and interrupts to be initialized.
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "adc.h"
+#include "gpio.h"
+#include "soft_timer.h"
+#include "status.h"
+
+// Conversion callbacks for ADCs.
+typedef uint16_t (*power_path_conversion_callback)(uint16_t value);
+
+typedef enum {
+  POWER_PATH_SOURCE_ID_DCDC = 0,
+  POWER_PATH_SOURCE_ID_AUX_BAT,
+  NUM_POWER_PATH_SOURCE_ID,
+} PowerPathSourceID;
+
+// For storing voltage and current values.
+typedef struct PowerPathVCReadings {
+  uint16_t voltage;
+  uint16_t current;
+} PowerPathVCReadings;
+
+// For storing the Power Path Source (eg AUX Battery vs DCDCs).
+typedef struct PowerPathSource {
+  const PowerPathSourceID id;
+  const GPIOAddress uv_ov_pin;
+  const GPIOAddress voltage_pin;
+  const GPIOAddress current_pin;
+  volatile PowerPathVCReadings readings;
+  power_path_conversion_callback current_convert;
+  power_path_conversion_callback voltage_convert;
+  uint32_t period_us;
+  SoftTimerID timer_id;
+  bool monitoring_active;
+} PowerPathSource;
+
+// For storing the power path configuration.
+typedef struct PowerPathCfg {
+  const GPIOAddress enable_pin;
+  const GPIOAddress shutdown_pin;
+  PowerPathSource aux_bat;
+  PowerPathSource dcdc;
+} PowerPathCfg;
+
+// Configures the GPIO pins for the power path.
+StatusCode power_path_init(PowerPathCfg *pp);
+
+// Enables the power path.
+StatusCode power_path_enable(const PowerPathCfg *pp);
```

Comment:
What does enabling the power path mean? I thought it was all in hardware?

Reply:
@taipingli  connected these to the MCU, according to the data sheet they are internally pulled up but can/should be pulled up externally as well if they are connected to something. I just pulled these high more as a safety than anything. I'm happy to move this into the `init` function as well but since this is config external to the MCU I felt it should be a separate function. Thoughts?

Reply:
Hmm, I think it would make sense to move it to the init function. I'm not sure there's much value in controlling the powerpath since the MCU will die if it's disabled. :)

Reply:
Done.